include/NovelRT/Utilities/Memory.h

```diff
@@ -0,0 +1,48 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#ifndef NOVELRT_UTILITIES_MEMORY_H
+#define NOVELRT_UTILITIES_MEMORY_H
+
+#include <string.h>
+#include <cassert>
+#include <cstring>
+#include <NovelRT/Exceptions/Exceptions.h>
+
+namespace NovelRT::Utilities
+{
+    class Memory
+    {
+    public:
+        static void Copy(void* destination, size_t destinationSize, const void* source, size_t count)
```

Comment:
Based on the usage I'm seeing, does it make sense to either remove the `destinationSize` parameter (and assume `count` == `size`? Or an overload that would do as such? (In a way that mimics `memcpy` over `memcpy_s`)

Reply:
The intent is to mimic memcpy_s on all platforms where it's not available. This is why we added the destinationSize even though due to asserts its usage was compiled out in non Windows builds.
Due to the change outlined above that should not be an issue anymore and we actually are checking against it. Keep in mind this is most likely just the initial implementation and as my knowledge grows I will extend the Memory utilities that we want to use in our own engine.